src/js/apps/globals/nav_app.js

```diff
@@ -216,10 +221,54 @@ export default App.extend({
       'destroy'() {
         navView.triggerMethod('search:active', false);
       },
+      'show:addPatient': this.showAddPatientModal,
     });
 
     Radio.request('modal', 'show:custom', patientSearchModal);
 
     navView.triggerMethod('search:active', true);
   },
+  showAddPatientModal() {
+    const storedAddPatientModal = this.getState('addPatientModal');
+    let addPatientView;
+
+    if (storedAddPatientModal) {
+      addPatientView = new AddPatientModal({
+        model: storedAddPatientModal.model,
+        state: storedAddPatientModal.getState().attributes,
+      });
+    } else {
+      addPatientView = new AddPatientModal({
+        model: Radio.request('entities', 'patients:model', {}),
+      });
+    }
+
+    const addPatientModal = Radio.request('modal', 'show:custom', addPatientView);
+
+    this.setState('addPatientModal', null);
+
+    this.listenTo(addPatientModal, {
+      'save'({ model }) {
+        model.saveAll(model.attributes)
+          .then(({ data }) => {
+            Radio.trigger('event-router', 'patient:dashboard', data.id);
+            addPatientModal.destroy();
+          })
+          .fail(({ responseJSON }) => {
+            // This assumes that only the similar patient error is handled on the server
+            addPatientView.setState({
+              errors: {
+                name: 'duplicate',
```

Comment:
This may not be ideal, I'd assume we want to surface BE errors as they are. But to do that here, we'll need a BE error state and a FE error state so that we can differentiate the errors message string from the BE and the error code string from the FE.